src/components/Login/LoginForm.jsx

```diff
@@ -1,15 +1,51 @@
 import "./LoginForm.css";
+import { useState } from "react";
 import { Link } from "react-router-dom";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faEyeSlash, faEye } from "@fortawesome/free-solid-svg-icons";
+import { useLoginHandler } from "../../common/helpers/Auth/LoginHandler";
+import { FormError } from "../FormError/FormError";
+import { useFormError } from "../../common/context/Form-Error-Context";
 
 const LoginForm = () => {
+    const [showPassword, setShowPassword] = useState(false);
+    const { loginHandler, loginAsGuestHandler, loginFormData, setLoginFormData } = useLoginHandler();
+    const { formDataErr, isFormError} = useFormError();
+
+    const handlePasswordDisplay = (e) => {
+        e.preventDefault();
+        showPassword ? setShowPassword(false): setShowPassword(true);
+    }
+
+    const formInputOnChangeHandler = (event) => {
+        setLoginFormData({...loginFormData, [event.target.name]: event.target.value });
+    }
+
     return (
     <main className="main-container">
         <div className="form-container centered-flex-col-container">
-            <form className="log-in-form centered-flex-col-container">
+            <form className="log-in-form centered-flex-col-container" onSubmit={(event) => loginHandler(event, loginFormData)}>
                 <h2 className="form-heading heading-2 space-S">Log In</h2>
-                <input className="input-primary login-input space-S" type="text" placeholder="Enter your email"/>
-                <input className="input-primary login-input space-S" type="text" placeholder="Enter your password"/>
-                <button className="btn btn-primary login-btn rounded-med space-M">Log In</button>
+                <div className="login-form-fields centered-flex-col-container">
+                    <label className="input-label">Username*</label>
```

Comment:
Add a for attribute to the label so that clicking it would focus the related input.

Reply:
Sure, thanks for pointing this out! 🙌